### PR TITLE
SHL-106: Java Configuration support

### PR DESCRIPTION
**Pull request by robinhowlett**
- Updated Bootstrap.java to accept basePackages String varargs and to include them in ClassPathBeanDefinitionScanner scan
- Updated sample HelloWorld project to demonstrate mixing XML and Java Configuration
- Updated docbook-reference-plugin dependency group ID and version

### CODE REVIEW

File: src/main/java/org/springframework/shell/Bootstrap.java [hunk 1]
```diff
@@ -37,25 +40,26 @@
...
-
-	private static StopWatch sw = new StopWatch("Spring Shell");
-
 	private static CommandLine commandLine;
+	private static StopWatch sw = new StopWatch("Spring Shell");
 
 	private GenericApplicationContext ctx;
 
-	public static void main(String[] args) throws IOException {
+	public static void main(String[] args, String... basePackages) throws IOException {
```
Comment (ericbottard) [inline src/main/java/org/springframework/shell/Bootstrap.java]: Can't change the signature of `public static void main(String[])` like that. Will need to find another way to pass those in

Comment (robinhowlett) [inline src/main/java/org/springframework/shell/Bootstrap.java]: Thanks for catching that mistake. I've updated to PR to revert back to a spec-compliant main method and to use a default package to scan (similar to how XML config is supported)

**Comment by ericbottard** [comment]
Closing outdated PRs and issues

**Comment by kevinm416** [comment]
I would like to use the feature implemented in this PR. Were there any outstanding requests that weren't addressed? Could this be merged as-is? Or maybe there's an alternative way to accomplish the same thing?

**Comment by raul-guerrero** [comment]
hi guys,
@ericbottard was this merged into master? will this feature be available?
Thanks!

**Comment by pivotal-issuemaster** [comment]
@robinhowlett Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-shell&pullRequestId=66)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-shell&pullRequestId=66) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@robinhowlett Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-shell&pullRequestId=66)!

**Comment by ericbottard** [comment]
Hi @raul-guerrero 
This feature was for Spring Shell 1.x
Spring Shell 2.x has out of the box `@Configuration` (and boot) support

**Pull request closed without merging**